lib/extensions/attachable/active_record/base.rb

```diff
@@ -27,6 +41,51 @@ def files=(files)
         attachment_references.build(file: file)
       end
     end
+
+    private
+
+    # Delete the attachment references which are not in the content.
+    def update_attachment_references
+      return if attachment_references.empty?
+
+      attachment_reference_ids = active_attachment_reference_ids
```

Comment:
Then, how would you know which IDs to keep?

Maybe it should be a callback that aggregates each of the `on:` fields, then combines them before assigning them all into the ids attribute